_P208_Nokia_LCD_5110.ino

```diff
@@ -271,12 +277,19 @@ boolean displayText3(char deviceTemplate[][48], struct EventStruct *event ){ //4
             lcd3->setTextSize(1);
           }
           String tmpString = deviceTemplate[x];
-          String newString = "";
+          String newString = " ";
+          String newString3 = " ";
           if (tmpString.length())
           {
             newString = parseTemplate(tmpString, false);
           }else{
             newString = html_input[x]; // if webformline is empty use html input
+            // Remove trailing spaces for the next time
+            std::string newString2 (newString.c_str()); 

```

Comment:
Please use Arduino's `String` with capital S and not `std::string`.
The Arduino String class has a function `trim()` to remove leading and trailing spaces.
Then you don't need to copy it over and over to new objects.

Reply:
Only the trailing spaces has to be removed. So I need to use ".find_last_not_of" method of std::string.
Alternative, working code, for removing trailing spaces in String is welcome.

Reply:
```c++
newString = html_input[x]; // if webformline is empty use html input
// Remove trailing spaces for the next time
while (newString.endsWith(' ')) {
  newString = newString.substring(0, newString.length() - 1);
}
```
Not the most efficient code, but it should strip all trailing spaces.

Reply:
thanks, I use you code. And "/0" is not needed now!